G2P.pm

```diff
@@ -679,7 +686,31 @@ sub gene_overlap_filtering {
   my $gene_stable_id = get_gene_stable_id($transcript);
   my $pass_gene_overlap_filter = $self->{g2p_gene_cache}->{$gene_stable_id};
   my @gene_xrefs = ();
-  if (! defined $pass_gene_overlap_filter) {
+  if (! defined $pass_gene_overlap_filter && ! defined $self->{user_params}->{filter_by_gene_symbol}) {
```

Comment:
If the user sets `filter_by_gene_symbol` to anything other than 1, this and other parts of the code will still not run, which the user may not expect.

An easy way to avoid this is to undefine `filter_by_gene_symbol` if it is not 1. This can be done by adding the following code to the `new` function (before its `return` statement, e.g. in [line 370](https://github.com/Ensembl/VEP_plugins/blob/456807bf9469995ce0ddaa37a43b365c8ccdc538/G2P.pm#L370)):
```perl
my $filter_by_gene_symbol = $self->{user_params}->{filter_by_gene_symbol};
$self->{user_params}->{filter_by_gene_symbol} = undef() if defined($filter_by_gene_symbol) and $filter_by_gene_symbol != 1;
```